modules/core/client/app/config.js

```diff
@@ -1,82 +1,103 @@
-    'ngSanitize',
-    'ngMessageFormat',
-    'angulartics',
-    'ui.router',
-    'ui.bootstrap.dateparser',
-    'ui.bootstrap.buttons',
-    'ui.bootstrap.collapse',
-    'ui.bootstrap.dropdown',
-    'ui.bootstrap.modal',
-    'ui.bootstrap.popover',
-    'ui.bootstrap.progressbar',
-    'ui.bootstrap.tabs',
-    'ui.bootstrap.tooltip',
-    'ui.bootstrap.typeahead',
-    'ui.bootstrap.datepicker',
-    'angularMoment',
-    'nemLogging',
-    'ui-leaflet',
-    'ngFileUpload',
-    'zumba.angular-waypoints',
-    'localytics.directives',
-    'angular-loading-bar',
-    'trTrustpass',
-    'angular-mailcheck',
-    'angular-locker',
-    'angular-confirm',
-    'angularGrid',
-  ];
-
-  /**
-   * Load different service dependency for Angulartics depending on environment
-   * @link https://github.com/angulartics/angulartics
-   */
-  if (appEnv === 'production') {
-    // @link https://github.com/angulartics/angulartics-google-analytics
-    appModuleVendorDependencies.push('angulartics.google.analytics');
-  } else if (appEnv === 'development') {
-    // @link https://github.com/angulartics/angulartics/blob/master/src/angulartics-debug.js
-    appModuleVendorDependencies.push('angulartics.debug');
-  } else {
-    // For "test" environment
-    // See `testutils/client/angulartics-null.testutil.js`
-    appModuleVendorDependencies.push('angulartics.null');
-  }
-
-  // Add a new vertical module
-  var registerModule = function (moduleName, dependencies) {
-    // Create angular module
-    angular.module(moduleName, dependencies || []);
-
-    // Add the module to the AngularJS configuration file
-    angular.module(appModuleName).requires.push(moduleName);
-  };
-
-  return {
-    appEnv: appEnv,
-    appModuleName: appModuleName,
-    appModuleVendorDependencies: appModuleVendorDependencies,
-    registerModule: registerModule,
-  };
-}());
-
-module.exports = AppConfig;
+// Init module configuration options
+// When testing, `window.env` is undefined, thus default to 'test'
+// eslint-disable-next-line angular/window-service
+const appEnv = process.env.NODE_ENV || 'test';
+
+const appModuleName = 'trustroots';
+const appModuleVendorDependencies = [
+  ngreact.name,
+  ngAria,
+  ngResource,
+  ngAnimate,
+  ngTouch,
+  ngSanitize,
+  ngMessageFormat,
+  'angulartics',
+  uiRouter,
+  angularMoment,
+  'nemLogging',
+  'ui-leaflet',
+  ngFileUpload,
+  'zumba.angular-waypoints',
+  'localytics.directives',
+  'angular-loading-bar',
+  trTrustpass,
+  'angular-mailcheck',
+  angularLocker,
+  'angular-confirm',
+  angularGrid,
+  ...bootstrapModules,
+];
+
+/**
+ * Load different service dependency for Angulartics depending on environment
+ * @link https://github.com/angulartics/angulartics
+ */
+if (appEnv === 'production') {
+  // @link https://github.com/angulartics/angulartics-google-analytics
+  appModuleVendorDependencies.push('angulartics.google.analytics');
+  require('angulartics-google-analytics/lib/angulartics-ga');
```

Comment:
This is fine like this and no changes needed; I just wanted to highlight Webpack dynamic imports which sometimes is a nifty feature for situations like this.

https://webpack.js.org/guides/code-splitting/#dynamic-imports

Reply:
Yup, actually, related, these should be chosen a way that works with webpack define plugin and javascript minification that removes the other branches in production... I suspect right now all 3 are included in the production build (but still the correct one used). I will leave that till after merge...

I think using a dynamic import here would risk the library not having loaded by the time angular is bootstrapped as they are async and return a promise (so execution continues below, before the import is complete), whereas traditional `require` should still be synchronous. Just basing things on my stored knowledge, could be wrong.

Reply:
Yup, bundle size was my concern as well and valid points about Angular. It's all code going away eventually, anyway, and not a ton of KBs.

I think I've seen some Webpack plugin that's smart about these env checks and can do the codesplitting. 🤔  I don't think it's necessarily worth setting up for this, tho.

Reply:
Splitting is built into webpack now, but currently we have no way to tell the server to serve files other than main.js/css, but I have a plan for that... I wrote about it somewhere else already.

The env checks work because the DefinePlugin literally rewrites the code so that (depending on how you've configured it), `process.env.NODE_ENV === 'test'` gets turned into `'production' === 'test' (for example), and a normal javascript minifier can do dead code elimination. I _think_ the way we have used the DefinePlugin will make this not work (`JSON.stringify`ing the whole of process.env), it's on my list to investigate what actually happens...